### ISSUE TITLE
x509 plugin can't send data to Influx, 1.16 working, 1.17 broken

### ISSUE THREAD

**Issue by loganmarchione** [influxdata/telegraf#8880]
Recently upgraded Telegraf from 1.16 to 1.17 and noticed the x509 input plugin is not working anymore. Rolling back to 1.16 fixes it.

### Relevant telegraf.conf:

```toml
[[inputs.x509_cert]]
  sources = ["https://10.10.1.9:5001", "https://10.10.1.1:443"]
```

### System info:

OS: Ubuntu 20.04.2
Docker: 20.10.3

Running everything in containers:
InfluxDB: influxdb:1.8-alpine
Working version: telegraf:1.16-alpine
Broken version: telegraf:1.17-alpine

### Docker

This is relevant snippet of the `docker-compose.yml` file.
```
  certcheck:
    container_name: monitoring_certcheck
    image: telegraf:1.17-alpine
    restart: unless-stopped
    networks:
      - certcheck
    volumes:
      - 'certcheck_config:/etc/telegraf:ro'
```

### Steps to reproduce:

1. ...
2. ...

### Expected behavior:

Here is the log of it working on 1.16.
```
2021-02-19T03:00:06Z D! [agent] Attempting connection to [outputs.influxdb]
2021-02-19T03:00:06Z D! [agent] Successfully connected to outputs.influxdb
2021-02-19T03:00:06Z D! [agent] Starting service inputs
2021-02-19T03:00:16Z D! [outputs.influxdb] Wrote batch of 3 metrics in 15.039098ms
2021-02-19T03:00:16Z D! [outputs.influxdb] Buffer fullness: 0 / 10000 metrics
```

### Actual behavior:

Here is the log of it not working on 1.17.
```
2021-02-19T02:57:26Z D! [agent] Attempting connection to [outputs.influxdb]
2021-02-19T02:57:26Z D! [agent] Successfully connected to outputs.influxdb
2021-02-19T02:57:26Z D! [agent] Starting service inputs
2021-02-19T02:57:30Z E! [inputs.x509_cert] Error in plugin: cannot get SSL cert 'https://10.10.1.9:5001': dial tcp 10.10.1.9:5001: i/o timeout
2021-02-19T02:57:30Z E! [inputs.x509_cert] Error in plugin: cannot get SSL cert 'https://10.10.1.1:443': dial tcp 10.10.1.1:443: i/o timeout
2021-02-19T02:57:36Z D! [outputs.influxdb] Buffer fullness: 0 / 10000 metrics
```

**Comment by viperstars** [comment]
Add timeout = "10s" can fix this temporarily. The fix PR has been merged to master.

**Comment by loganmarchione** [comment]
In the output section?
```
###############################################################################
#                            OUTPUT PLUGINS                                   #
###############################################################################

[[outputs.influxdb]]
  urls = ["http://influxdb:8086"]

  database = "my_database"

  skip_database_creation = true

  retention_policy = ""

  timeout = "10s"

  username = "username"
  password = "password"
```

I'm still getting the `dial tcp: i/o timeout` error.

**Comment by viperstars** [comment]
> In the output section?
> 
> ```
> ###############################################################################
> #                            OUTPUT PLUGINS                                   #
> ###############################################################################
> 
> [[outputs.influxdb]]
>   urls = ["http://influxdb:8086"]
> 
>   database = "my_database"
> 
>   skip_database_creation = true
> 
>   retention_policy = ""
> 
>   timeout = "10s"
> 
>   username = "username"
>   password = "password"
> ```
> 
> I'm still getting the `dial tcp: i/o timeout` error.

 Add timeout = "10s" in the input section

**Comment by helenosheaa** [comment]
@loganmarchione the fix was merged in our latest patch release so on 1.17.3 it should be fixed
